rpc/src/test/scala/org/bitcoins/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,1713 @@
+package org.bitcoins.rpc
+
+import java.io.File
+import java.net.URI
+import java.util.Scanner
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import org.bitcoins.core.crypto.{DoubleSha256Digest, ECPrivateKey, ECPublicKey}
+import org.bitcoins.core.currency.{Bitcoins, Satoshis}
+import org.bitcoins.core.protocol.transaction.{
+  Transaction,
+  TransactionInput,
+  TransactionOutPoint
+}
+import org.bitcoins.core.util.BitcoinSLogger
+import org.bitcoins.rpc.client.{RpcClient, RpcOpts}
+import org.scalatest.{AsyncFlatSpec, BeforeAndAfter, BeforeAndAfterAll}
+import org.bitcoins.core.number.{Int64, UInt32}
+import org.bitcoins.core.protocol.{BitcoinAddress, P2PKHAddress}
+import org.bitcoins.core.protocol.script.{
+  P2SHScriptSignature,
+  ScriptPubKey,
+  ScriptSignature
+}
+import org.bitcoins.core.wallet.fee.SatoshisPerByte
+import org.bitcoins.rpc.jsonmodels.{
+  GetBlockWithTransactionsResult,
+  GetTransactionResult,
+  RpcAddress
+}
+
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration.DurationInt
+import scala.util.Try
+
+class RpcClientTest
+    extends AsyncFlatSpec
+    with BeforeAndAfterAll
+    with BeforeAndAfter {
+  implicit val system = ActorSystem()
+  implicit val m = ActorMaterializer()
+  implicit val ec = m.executionContext
+  implicit val networkParam = TestUtil.network
+
+  private var newPort = networkParam.port + 5
```

Comment:
Let's not use `var` here. Just create method that generates a random port like [this](https://github.com/bitcoin-s/bitcoin-s-rpc-client/blob/master/src/test/scala/org/bitcoins/rpc/util/TestUtil.scala#L74)

Reply:
if you need to save the port, save it in a `val` where ever you are using it. In general we shouldn't use `var` unless absolutely needed. It creates nightmares for concurrency